### ISSUE TITLE
[Improvement]: cursor based pagination in listings

### ISSUE THREAD

**Issue by dkarlovi** [pimcore/pimcore#13470]
### Improvement description

When you're paging through a large dataset (maybe the entire dataset), typical [`LIMIT/OFFSET` approach doesn't work](https://explainextended.com/2009/10/23/mysql-order-by-limit-performance-late-row-lookups/):

> `LIMIT   150000, 10` (...) can be reformulated like this: take the last 150,010 rows in id order and return the first 10 of them

There's numerous references about the same topic, how to solve it, for example:
https://shopify.engineering/pagination-relative-cursors

The proposal here is to add a cursor-based API to Listing to allow streaming through the pages efficiently.

# Alternatives

There's [some tricks](https://stackoverflow.com/a/4502426/672885) which are meant to speed up the process, but I didn't test them.

**Comment by jdreesen** [comment]
Here is an article on the subject that I found interesting: https://aaronfrancis.com/2022/efficient-pagination-using-deferred-joins

**Comment by dkarlovi** [comment]
Yes, that's basically the hack mentioned above. Maybe it's worth a shot?
